src/coreComponents/common/format/table/TableFormatter.cpp

```diff
@@ -96,315 +134,525 @@ string buildCell( TableLayout::Alignment const alignment, string_view value, int
+                                                   string & separatorLine ) const
+{
+  std::string const tableTitle = std::string( tableLayout.getTitle() );
+  size_t const margins = (size_t) tableLayout.getBorderMargin() * 2;
+  size_t const horizontalBar = 2;
 
-    std::string const leftBorder = GEOS_FMT( "{}{:-<{}}", m_horizontalLine, "", borderMargin );
-    std::string const rightBorder = GEOS_FMT( "{}{:-<{}}", m_horizontalLine, "", borderMargin );
-    std::string const columnJoin = stringutilities::join( maxStringPerColumn, patternBetweenColumns );
+  size_t sectionlineLength = 0;
+  size_t nbColumns = 0;
+  size_t nbHiddenColumns = 0;
 
-    std::ostringstream oss;
-    oss << leftBorder << columnJoin << rightBorder;
-    sectionSeparatingLine = oss.str();
+  for( auto const & column : cellsHeaderLayout[0] )
+  {
+    if( column.m_cellType == CellType::Hidden )
+    {
+      nbHiddenColumns++;
+    }
+    if( column.m_cellType != CellType::Hidden && column.m_cellType != CellType::MergeNext )
+    {
+      sectionlineLength += column.m_cellWidth;
+      nbColumns++;
+    }
   }
 
-  { // top line separator
-    // -2 because we can have '+' to the extremity of top separator
-    integer const topSeparatorLength = sectionSeparatingLine.size() - 2;
-    topSeparator = GEOS_FMT( "{}{:-<{}}{}", m_horizontalLine, "", topSeparatorLength, m_horizontalLine );
+  size_t const spacingBetweenColumns = (nbColumns - 1) * (size_t) tableLayout.getColumnMargin();
+  sectionlineLength += spacingBetweenColumns + margins + horizontalBar;
+
+  size_t titleRowLength = tableTitle.length() + margins + horizontalBar;
+  size_t maxLength = std::max( {titleRowLength, sectionlineLength} );
+  if( sectionlineLength < maxLength )
+  {
+    size_t const paddingCharacters = maxLength - sectionlineLength;
+    adjustColumnWidths( cellsHeaderLayout, nbHiddenColumns, paddingCharacters );
+    adjustColumnWidths( cellsDataLayout, nbHiddenColumns, paddingCharacters );
+    sectionlineLength = maxLength;
   }
+
+  separatorLine = GEOS_FMT( "{:-^{}}", m_horizontalLine, sectionlineLength );
 }
 
-void TableTextFormatter::outputTopRows( std::ostringstream & tableOutput,
-                                        std::vector< string > const & msg,
-                                        string_view topSeparator,
-                                        TableLayout::Alignment alignment ) const
+void TableTextFormatter::adjustColumnWidths( CellLayoutRows & cells,
+                                             size_t nbHiddenColumns,
+                                             size_t const paddingCharacters ) const
 {
-  if( msg.size() != 0 && msg[0] != "" )
+  size_t const numRows = cells.size();
+  size_t const nbColumns = cells[0].size();
+
+  size_t remainingPaddingForLastColumn = paddingCharacters % (nbColumns - nbHiddenColumns);
+  size_t paddingPerColumn = std::floor( paddingCharacters / (nbColumns - nbHiddenColumns));
```

Comment:
`floor()` useless on integers, be advised that integer division already round its result toward `0` in order to return an integer result.

Reply:
This was very _enfoncer des portes ouvertes_.